### DISCUSSION TITLE
Metadata

### DISCUSSION THREAD

**Discussion by mkutu** [precision-sustainable-ag/SemiF-AnnotationPipeline#6] [General]
Here's a place for general discussion about semifield metadata and what we think would be important information to collect. 

Metadata could be used to identifying environmental conditions like wind, cloudiness, and moisture that may have impact on annotation quality.

As of know a running list of metadata items that we may or may not think are useful (and are not automatically embedded in exif data include:

- What species are on table (to validate prediction) - this may not be necesssary as we might be able to validate from pictures alone, if we are good weed identifiers :)
- a unique upload id - generated automatically
- location - gathered from unique qr-codes
- cloud cover - may or may not have bearing on image quality snese we are using flash. 
-  camera height - I suspect camera height maybe slightly different among 3 locations. Would be able to sort using this.
- RH
- Wind - The flash tends to overheat when doing shutter speeds faster than 1/200 seconds
- Moisture - moisture might lead to specular reflections that would be nice to have automatically sorted
- Moisture (humidity) -

**Comment by skovsens** [comment]
Good points! What is RH in this context? 
Some additional metadata below:
Need to have:
- Measured-once metadata includes accurate measurement (in three axis) of each of the QR-markers. While this is a little tricky, (especially vertically) we can likely refine measurements using the autoSfM script.
- For color calibration purposes, it would make sense to make a reserved spot for the calibration chart? Ideally in the initial part of the image collection, so we don't rely on autoSfM to find the right image (since autoSfM should be performed after the color calibration process).

Need to check on each camera:
- Correct time settings  (To make sure that we can find corresponding metadata for each image)
- Correct file name conventions describing the site/state
- Correct image format (non-compressed raw and extra-fine jpeg) and full-frame sensor format
- Correct exposure and flash settings
- Remove finger prints from lens front and dust from the image sensor

Nice to have:
- If the weather station supports it: Solar radiation. In case of high fluctuations during the day, we might want to look into those image calibrations.
- If the weather station supports it: (Recent or ongoing) Precipitation 
- (Cumulative) temperature. Some scientists relate cumulative temperature (Celsius) with plant/biomass growth. This might be relevant at some point for purposes other than weed recognition.

**Comment by skovsens** [comment]
With regards to coupling all the levels of metadata together, I imagine that we need some ID's for each step of the way. That could look something like this:
Drafted metadata hierarchy
	Per site:
		Site-ID
		QR-marker location
		species location
	Per collection:
		Collection-ID
		Weather data
		Color calibration
	Per image
		Image-ID
		Focal length
		Bounding box plant annotations
	Per data collection processing
		Processing-ID
		Bounding box plant annotations
	Per Bounding box annotation
		Bounding box-ID
		Corresponding image
		Plant species
		Plant mask
		Plant ID
	Per unique plant
		Plant ID
		List of all images
		List of all bounding box image crops
		List of all segmentations
		Plant species
		Seed date
		Emergence date
